x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/CronnableSchedule.java

```diff
@@ -37,7 +32,15 @@ public long nextScheduledTimeAfter(long startTime, long time) {
         assert time >= startTime;
         long nextTime = Long.MAX_VALUE;
         for (Cron cron : crons) {
-            nextTime = Math.min(nextTime, cron.getNextValidTimeAfter(time));
+            long nextValidTimeAfter = cron.getNextValidTimeAfter(time);
```

Comment:
Im curious, does the comparator not put these at either the begin or end of the sorted list here? should we just short circuit the loop once we find one of these outliers?

Reply:
the cron comparator is a string based comparator (which I don't think makes a ton of sense in context of a cron expression, maybe we can just remove it), and thus the sorting could be wrong in regards to finding the next valid time.

Reply:
example: `0 6 9 1 1 ? 2020` and `1 5 9 1 1 ? 2020`. The sorting puts the first expression in front, but the second one is actually triggered first